keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlVerificationRepository.kt

```diff
@@ -175,6 +175,13 @@ class SqlVerificationRepository(
         .set(VERIFICATION_STATE.VERIFICATION_ID, verification.id)
         .onDuplicateKeyUpdate()
         .set(VERIFICATION_STATE.STATUS, status)
+        .run {
+          if (metadata.isNotEmpty()) {
+            set(VERIFICATION_STATE.METADATA, metadata)
+          } else {
+            this
+          }
+        }
```

Comment:
This field was never getting set on update before. Ideally I'd like to use MySQL's `json_merge` function here so that it updates the document but I'm not sure how to do that with JOOQ. It's moot at the moment since the only thing currently written into this column is the Orca task id when running a container test job.

Reply:
I'd add a reminder comment here about the semantics of an empty metadata object, to help a future reader.

This is one of the things I find frustrating about the interaction of docstrings and subclassing. The semantics of an empty metadata object has already been documented on the `VerificationRepository` interface, but those comments aren't visible in this file.